services/gitdiff/gitdiff.go

```diff
@@ -956,6 +956,7 @@ func parseHunks(curFile *DiffFile, maxLines, maxLineCharacters int, input *bufio
 	lastLeftIdx := -1
 	leftLine, rightLine := 1, 1
 
+	engine := db.GetEngine(db.DefaultContext)
```

Comment:
Really not a fan of letting the engine spill in to services - if we're gonna do this what's the point of the functions in models.

Reply:
The code was there. The code was accessing database directly via db.Xxx.

The new code is just more clear than before.

Adding new packages is not the task or purpose of this PR. This PR just makes a clean up.